From 8b6a9e14822aa850bb3f06b49e9d8942769145ac Mon Sep 17 00:00:00 2001
From: Toshiko YOSHIDA <toshiko.yoshida@toshiba.co.jp>
Date: Tue, 16 May 2017 16:50:56 +0900
Subject: [PATCH] Fix race condition in perl's ExtUtils::MakeMaker

Many thanks to Nathanael Rensen <nathanael at polymorpheus dot com>
for tracking it down and supplying the patch.

Has been reported upstream and the fix incorporated into a larger change
Perl-Toolchain-Gang/ExtUtils-MakeMaker#192

Signed-off-by: Toshiko YOSHIDA <toshiko.yoshida@toshiba.co.jp>
---
 cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm | 72 ++++++++++---------------
 1 file changed, 28 insertions(+), 44 deletions(-)

diff --git a/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm b/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm
index 4140432..c23230e 100644
--- a/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm
+++ b/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm
@@ -911,7 +911,7 @@ OTHERLDFLAGS = '.$ld_opt.$otherldflags.'
 INST_DYNAMIC_DEP = '.$inst_dynamic_dep.'
 INST_DYNAMIC_FIX = '.$ld_fix.'
 
-$(INST_DYNAMIC): $(OBJECT) $(MYEXTLIB) $(INST_ARCHAUTODIR)$(DFSEP).exists $(EXPORT_LIST) $(PERL_ARCHIVE) $(PERL_ARCHIVE_AFTER) $(INST_DYNAMIC_DEP)
+$(INST_DYNAMIC): $(OBJECT) $(MYEXTLIB) $(BOOTSTRAP) $(INST_ARCHAUTODIR)$(DFSEP).exists $(EXPORT_LIST) $(PERL_ARCHIVE) $(PERL_ARCHIVE_AFTER) $(INST_DYNAMIC_DEP)
 ');
     if ($armaybe ne ':'){
 	$ldfrom = 'tmp$(LIB_EXT)';
@@ -963,7 +963,7 @@ MAKE
 
     push @m, <<'MAKE';
 	$(CHMOD) $(PERM_RWX) $@
-	$(NOECHO) $(RM_RF) $(BOOTSTRAP)
+	$(NOECHO) $(RM_RF) $(INST_BOOT)
 	- $(CP_NONEMPTY) $(BOOTSTRAP) $(INST_BOOT) $(PERM_RW)
 MAKE
 
-- 
2.1.4

